### ISSUE TITLE
code optimization for sdmTMB scenario simulations

### ISSUE THREAD

**Issue by AngeliaMiller** [thefaylab/sseep-analysis#12]
Hi @gavinfay, 

The simulation loop for the sdmTMB change scenarios has been added through commit [278197b](https://github.com/thefaylab/sseep-analysis/commit/278197b3022fe5d22242aafabe53cc05b9272f12) under the analysis-testing branch

Below is the r object for the fall model. Let me know if I left out any other data you may need. 

Thanks! 

[code-optim-issue9.zip](https://github.com/thefaylab/sseep-analysis/files/13241726/code-optim-issue9.zip)

**Comment by gavinfay** [comment]
@AngeliaMiller I was able to figure out from the commit where the fall model object should go, and which script to look at (going forward please include these info in your message).
However, there appears to be an inconsistency in the model object. 
Running line 46 of `01a-sim-fall-scenario.R` generates the below error:

<img width="601" alt="image" src="https://github.com/thefaylab/sseep-analysis/assets/8093556/e34ce2fd-ae47-4232-aa0f-16b5d782ae93">

**Comment by gavinfay** [comment]
What version of TMB are you using?

**Comment by AngeliaMiller** [comment]
I have sdmTMB 0.3.0 and TMB 1.9.6

**Comment by gavinfay** [comment]
Yup - inconsistency in our versions of sdmTMB was the culprit. I wound things forward (and then back) and it seems to be working now.

**Comment by gavinfay** [comment]
Note the code is still relying on the `fall_preds` object.

**Comment by gavinfay** [comment]
Looks like the time culprit is in generating the multivariate normals. Your solution here would be to reduce the dimension of the variance-covariance matrix (i.e. fewer knots), run things in parallel (you have multiple cores), or ask your advisor to buy a faster machine.
